### PR TITLE
Data samples order

### PR DESCRIPTION
**Pull request by jmorel**
Removes the `--data-samples-path` arg that was unused by the backend and was expecting the folder in which all data samples would be and adds `--data-sample-paths` which is a list of the path to each individual data sample folder.

Train and test method then load the data samples in the order specified by the new arg.

Once merged, there will need to be a new release, probably 0.7.0 for substra-tools, and we'll need to have a new release for the backend as well.

Related PRs:
* https://github.com/SubstraFoundation/substra-backend/pull/344
* https://github.com/SubstraFoundation/substra-tests/pull/155
* https://github.com/SubstraFoundation/substra/pull/236

The PRs need to be merged in the following order:
1. substra-tools, with the release to match.
2. substra-backend and substra-tests
3. substra (once the compatibility table is updated).

### CODE REVIEW

File: substratools/workspace.py [hunk 1]
```diff
@@ -34,29 +34,38 @@ def __init__(self, dirpath=None):
...
             self._get_default_path(DEFAULT_INPUT_PREDICTIONS_PATH)
 
         self.output_predictions_path = output_predictions_path or \
             self._get_default_path(DEFAULT_OUTPUT_PREDICTIONS_PATH)
 
-        dirs = [
-            self.input_data_folder_path,
-        ]
+        dirs = []
+        dirs.extend(self.input_data_folder_paths)
```
Comment (Esadruhn) [inline substratools/workspace.py:68]: why not `dirs = self.input_data_folder_paths` ?

**Review by Esadruhn — APPROVED** [review]
Good for me !

File: substratools/workspace.py (hunk 1 above)
Comment (Esadruhn) [reply substratools/workspace.py:68]: (style question, not blocking)

**Review by samlesu — APPROVED** [review]
LGTM

**Pull request merged**